sympy/sets/fancysets.py

```diff
@@ -903,7 +903,7 @@ def _union(self, other):
             elif self.polar and other.polar:
                 return ComplexRegion(Union(self.sets, other.sets), polar=True)
 
-        if other.is_subset(S.Reals):
+        if self == S.Complexes:
```

Comment:
I was thinking that may be moving this to `Complexes` would be better(?). i.e `S.Complexes` having its own `.union` or `._union` (whatever)